packages/interface/src/store/actions.js

```diff
@@ -89,11 +96,18 @@ export function unpinItem( scope, itemId ) {
  * @param {string} scope       The feature scope (e.g. core/edit-post).
  * @param {string} featureName The feature name.
  */
-export function toggleFeature( scope, featureName ) {
-	return function ( { select, dispatch } ) {
-		const currentValue = select.isFeatureActive( scope, featureName );
-		dispatch.setFeatureValue( scope, featureName, ! currentValue );
-	};
+export function* toggleFeature( scope, featureName ) {
+	deprecated( `dispatch( 'core/interface' ).toggleFeature`, {
+		since: '6.0',
+		alternative: `dispatch( 'core/preferences' ).toggleFeature`,
+	} );
+
+	yield controls.dispatch(
+		preferencesStore.name,
+		'toggleFeature',
+		scope,
+		featureName
+	);
 }
```

Comment:
I'm not sure how far to go with deprecations for the interface package.

It is documented as an experimental package. 

It doesn't seem possible to access it on the `wp` global (`wp.interface` ), but the actions and selectors are all available via `wp.data`, so I've chosen to deprecate those. 

I've deleted the components completely from the package, but open to advice on whether I should also deprecate those.